modules/drivers/mongo/src/metabase/driver/mongo/util.clj

```diff
@@ -89,6 +88,64 @@
     :else                         (throw (Exception. (str "with-mongo-connection failed: bad connection details:"
                                                           (:details database))))))
 
+(defn- connect-via-uri
+  [^MongoClientURI uri ^MongoClient conn]
+  (if-let [dbName (.getDatabase uri)]
+    (.getDB conn dbName)
+    (throw (IllegalArgumentException. "No database name specified in URI. Monger requires a database to be explicitly configured."))))
```

Comment:
Some of our really old code doesn't have this, so feel free to fix it when you come across it.

To i18n stuff you need to wrap it in `trs` (for system-level stuff, e.g. log messages) or `tru` (for user-facing messages). This would be `tru` because it would be displayed in the DB setup panel when you tried to connect a DB if it failed.

The cannonical way to import `tru` and `trs` is `(:require [metabase.util.i18n :refer [tru trs])`. Some older code is using a a puppetlabs namespace and you can change that when you see it

Caveats: 

*  the `tru` and `trs` macros expand to some i18n record types we're using and you can't pass them directly to Java Exception constructors. So you have to wrap them in a call to `(str)`. That should probably just be part of the macro anyway so you don't have to remember it, but the guy who wrote didn't do it that way because it made it hard to do runtime translation of our Settings descriptions. But that can be fixed
*  it all has to be one line so you because of the way the lib we're using works. (I don't know why this is the case. I am planning on forking that lib and fixing that and my other complaints about it when I get a chance.)

Reply:
Done